### ISSUE TITLE
NC 17.01 - Setup new SMB shares fails while old shares working properly till you change values, even if they are the same

### ISSUE THREAD

**Issue by HaStub** [nextcloud/server#18111]
### Steps to reproduce
it is reproduceable, and testend on 2 instances

### Expected behaviour
When saving changes on a external share setup e.g. on /settings/user/externalstorages a new oder changed share fails to connect while old shares working properly

### Actual behaviour
settings are saved but the connection could not be established

### Server configuration

**Operating system**:
debian 9.11

**Web server:**
nginx 1.10.3

**Database:**
MariaDB 10.1.41

**PHP version:**
7.3.11

**Nextcloud version:** (see Nextcloud admin page)
17.01

** Samba Version:**
Version 4.5.16-Debian

**Updated from an older Nextcloud/ownCloud or fresh install:**
yes, old stable, guess 16.06

**Where did you install Nextcloud from:**

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
  - accessibility: 1.3.0
  - activity: 2.10.1
  - admin_audit: 1.7.0
  - announcementcenter: 3.6.1
  - apporder: 0.8.0
  - audioplayer: 2.8.4
  - bookmarks: 2.3.1
  - bruteforcesettings: 1.4.0
  - calendar: 1.7.1
  - camerarawpreviews: 0.7.2
  - carnet: 0.18.11
  - checksum: 0.4.3
  - circles: 0.17.10
  - cloud_federation_api: 1.0.0
  - comments: 1.7.0
  - contacts: 3.1.6
  - data_request: 1.4.0
  - dav: 1.13.0
  - deck: 0.7.0
  - drawio: 0.9.4
  - dropit: 0.3.0
  - event_update_notification: 1.0.0
  - extract: 1.2.2
  - federatedfilesharing: 1.7.0
  - federation: 1.7.0
  - files: 1.12.0
  - files_accesscontrol: 1.7.0
  - files_automatedtagging: 1.7.0
  - files_downloadactivity: 1.6.0
  - files_external: 1.8.0
  - files_frommail: 0.3.0
  - files_markdown: 2.1.0
  - files_mindmap: 0.0.15
  - files_pdfviewer: 1.6.0
  - files_readmemd: 1.1.2
  - files_retention: 1.6.0
  - files_rightclick: 0.15.1
  - files_sharing: 1.9.0
  - files_trackdownloads: 1.6.0
  - files_trashbin: 1.7.0
  - files_versions: 1.10.0
  - files_videoplayer: 1.6.0
  - firstrunwizard: 2.6.0
  - gallery: 18.4.0
  - gpxedit: 0.0.12
  - gpxmotion: 0.0.10
  - gpxpod: 4.1.0
  - group_everyone: 0.1.3
  - groupquota: 0.1.1
  - guests: 1.3.1
  - impersonate: 1.4.0
  - keeporsweep: 0.2.1
  - logreader: 2.2.0
  - lookup_server_connector: 1.5.0
  - mail: 0.19.0
  - metadata: 0.10.0
  - music: 0.11.1
  - news: 14.0.1
  - nextcloud_announcements: 1.6.0
  - notes: 3.0.3
  - notifications: 2.5.0
  - oauth2: 1.5.0
  - ocsms: 2.1.6
  - ownbackup: 19.9.0
  - passman: 2.3.5
  - password_policy: 1.7.0
  - phonetrack: 0.5.11
  - polls: 0.10.4
  - previewgenerator: 2.1.0
  - privacy: 1.1.0
  - provisioning_api: 1.7.0
  - quickaccesssorting: 1.0.2
  - quicknotes: 0.2.1
  - quota_warning: 1.6.0
  - ransomware_protection: 1.5.1
  - recommendations: 0.5.0
  - registration: 0.4.7
  - richdocuments: 3.4.4
  - scanner: 0.1.7
  - serverinfo: 1.7.0
  - sharebymail: 1.7.0
  - sharerenamer: 2.7.2
  - smb_test: 0.2.3
  - social: 0.2.101
  - socialsharing_diaspora: 1.0.5
  - socialsharing_email: 1.0.6
  - socialsharing_facebook: 1.0.5
  - socialsharing_twitter: 1.0.5
  - spreed: 7.0.2
  - support: 1.0.1
  - survey_client: 1.5.0
  - systemtags: 1.7.0
  - talk_simple_poll: 1.0.0
  - tasks: 0.11.3
  - terms_of_service: 1.3.1
  - text: 1.1.1
  - theming: 1.8.0
  - timetracker: 0.0.34
  - twofactor_backupcodes: 1.6.0
  - twofactor_gateway: 0.14.1
  - twofactor_nextcloud_notification: 2.1.1
  - twofactor_totp: 4.0.0
  - twofactor_u2f: 4.0.0
  - updatenotification: 1.7.0
  - user_external: 0.7.0
  - user_saml: 2.4.0
  - user_usage_report: 1.1.2
  - video_converter: 0.1.2
  - viewer: 1.2.0
  - workflow_pdf_converter: 1.2.0
  - workflowengine: 1.7.0
Disabled:
  - afterlogic
  - dashboard
  - dicomviewer
  - drop_account
  - encryption
  - files_ebookreader
  - files_external_gdrive
  - files_fulltextsearch
  - files_inotify
  - files_photospheres
  - forms
  - fulltextsearch
  - keeweb
  - ojsxc
  - ownpad
  - passwords
  - radio
  - sociallogin
  - user_ldap
  - weather
  - zenodo
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.*.org",
            "cloud-001.*.org",
            "cloud-001.fritz.box"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "17.0.1.1",
        "overwrite.cli.url": "https:\/\/cloud-001",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "default_language": "de",
        "default_locale": "de_DE",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": "",
        "loglevel": 2
    }
}

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
local and smb
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
### Client configuration
**Browser:**
Firefox, Chrome, Brave, Edge
**Operating system:**
Windows 10
#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
  GNU nano 2.7.4                                                                    File: nextcloud.log
no errors 
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by ex9r** [comment]
same issue.. been debugging for days now, without success. 
CLI smbclient connects just fine, NC shows a red exclamation mark

**Comment by kesselb** [comment]
Duplicate of #17846

Will be fixed with 17.0.2 and 16.0.7
